mllib/src/main/scala/org/apache/spark/ml/evaluation/MultilabelClassificationEvaluator.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.evaluation
+
+import org.apache.spark.annotation.{Experimental, Since}
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.param.shared._
+import org.apache.spark.ml.util._
+import org.apache.spark.mllib.evaluation.MultilabelMetrics
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+
+
+/**
+ * :: Experimental ::
+ * Evaluator for multi-label classification, which expects two input
+ * columns: prediction and label.
+ */
+@Since("3.0.0")
+@Experimental
+class MultilabelClassificationEvaluator (override val uid: String)
+  extends Evaluator with HasPredictionCol with HasLabelCol
+    with DefaultParamsWritable {
+
+  import MultilabelClassificationEvaluator.supportedMetricNames
+
+  def this() = this(Identifiable.randomUID("mlcEval"))
+
+  /**
+   * param for metric name in evaluation (supports `"f1Measure"` (default), `"subsetAccuracy"`,
+   * `"accuracy"`, `"hammingLoss"`, `"precision"`, `"recall"`, `"precisionByLabel"`,
+   * `"recallByLabel"`, `"f1MeasureByLabel"`, `"microPrecision"`, `"microRecall"`,
+   * `"microF1Measure"`)
+   * @group param
+   */
+  final val metricName: Param[String] = {
+    val allowedParams = ParamValidators.inArray(supportedMetricNames)
+    new Param(this, "metricName", "metric name in evaluation " +
+      s"${supportedMetricNames.mkString("(", "|", ")")}", allowedParams)
+  }
+
+  /** @group getParam */
+  def getMetricName: String = $(metricName)
+
+  /** @group setParam */
+  def setMetricName(value: String): this.type = set(metricName, value)
+
+  setDefault(metricName -> "f1Measure")
+
+  final val label: DoubleParam = new DoubleParam(this, "label",
+    "The label whose metric will be computed in precisionByLabel|recallByLabel|" +
+      "f1MeasureByLabel. Must be >= 0. The default value is 0.",
+    ParamValidators.gtEq(0.0))
+
+  /** @group getParam */
+  def getLabel: Double = $(label)
+
+  /** @group setParam */
+  def setLabel(value: Double): this.type = set(label, value)
```

Comment:
Hm, I think using `label` here is confusing, as this also has `labelCol` from `HasLabelCol` to specify the (multi-)label column. I wonder if we can just pick a clearly different name, like `metricClass` or `oneVsAllLabelCol`, that kind of thing?

Reply:
I perfer `metricClass` to `oneVsAllLabelCol`.
Or what about this?
```
def setClass(value: Double): this.type = set(clazz, value)
```

Reply:
@srowen Just found that in  [LogisticRegressionSummary](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala#L1404) per-class metrics are named `xxByLabel`, should we keep in line with it?

Reply:
Yeah, 'class' would be inconsistent with 'label'. I think we need to differentiate "the column that is the label" from "the label that is used to compute a per-label metric". Just "label" seems confusing. "metricLabel"? anything a little more descriptive.

Reply:
@srowen  If we use a name other than `label`, I personally perfer `metricClass` to `metricLabel` or others.

Reply:
Sounds good to me, just anything to distinguish its purpose. I suppose we say `labelCol` specifies the _class_ that the instance is labeled with, and `metricClass` chooses the _class_ to use when evaluating the 1 vs all metric.